frontend/src/metabase/components/SaveQuestionForm/context.tsx

```diff
@@ -77,6 +77,7 @@ export const SaveQuestionProvider = ({
   const [isSavedQuestionInitiallyChanged] = useState(
     isNotNull(originalQuestion) &&
       originalQuestion.type() !== "model" &&
+      originalQuestion.type() !== "metric" &&
```

Comment:
idk about this change, could you please explain it?

Reply:
We want to not show the `Save as new question / replace existing question` UI for metrics and models. So we check for that here.

I agree that it was confusing that this check was part of `isSavedQuestionInitiallyChanged` so I moved the two checks out of the `useState` for more clarity. Hope that makes it clearer?